merge/merger.go

```diff
@@ -30,8 +31,77 @@ func NewMerger(fhirHost string) *Merger {
 // is successful a new FHIR Bundle containing the merged patient record is returned.
 // If a merge fails, a FHIR Bundle containing one or more OperationOutcomes is
 // returned detailing the merge conflicts.
-func (m *Merger) Merge(source1, source2 string) (mergeID string, outcome *models.Bundle, err error) {
-	return mockMerge(source1, source2)
+func (m *Merger) Merge(source1ID, source2ID string) (mergeID string, outcome *models.Bundle, err error) {
+
+	var isMatched bool
+
+	source1 := new(models.Bundle)
+	source2 := new(models.Bundle)
+
+	//Unmarshal JSON to source objects
+	populateBundle(m.fhirHost+"/Bundle/"+source1ID, source1)
+	populateBundle(m.fhirHost+"/Bundle/"+source2ID, source2)
+
+	//Target bundle
+	createdTarget := new(models.Bundle)
+
+	//var birthdate models.FHIRDateTime
+	given1, family1 := getKeyData(source1)
+	given2, family2 := getKeyData(source2)
+
+	//Compare key data of both sources
+	if given1 == given2 && family1 == family2 {
+		isMatched = true
+	} else {
+		isMatched = false
+	}
+
+	createdTarget.ResourceType = "Bundle"
+
+	newRequest := new(models.BundleEntryRequestComponent)
+	newRequest.Method = "POST"
+
+	//Decide which type of Resource to send to FHIR server in the Bundle
+	if !isMatched {
+
+		newName := make([]models.HumanName, 1)
+		newGivenName := make([]string, 1)
+		newGivenName[0] = given1
+		newFamilyName := make([]string, 1)
+		newFamilyName[0] = family1
+		newName[0].Given = newGivenName
+		newName[0].Family = newFamilyName
+		newRequest.Url = "Patient"
+
+		newResource := new(models.Patient)
+		newResource.Name = newName
+		newEntry := make([]models.BundleEntryComponent, 1)
+		newEntry[0].Resource = newResource
+		newEntry[0].Request = newRequest
+		createdTarget.Entry = newEntry
+
+	} else {
+		//newResource := new(models.OperationOutcome)
+		newEntry := make([]models.BundleEntryComponent, 1)
+		//newEntry[0].Resource = newResource
+		newEntry[0].Request = newRequest
+		//createdTarget.Resource = newResource
+		createdTarget.Entry = newEntry
+	}
+
+	createdTarget.Type = "transaction"
+
+	bytestr, err := json.Marshal(createdTarget)
+
+	resp, err := http.Post(m.fhirHost+"/Bundle", "application/json", bytes.NewBuffer(bytestr))
+	if err != nil {
+
+		fmt.Println("Could not reach server ", m.fhirHost)
+	}
+	targetBundle := new(models.Bundle)
+	json.NewDecoder(resp.Body).Decode(&targetBundle)
+	mergeID = targetBundle.Id
```

Comment:
The way I designed it, the `mergeID` is not the same as the target bundle's ID. When we save merge state it's represented as follows:

```json
{
  "_id": "mergeID",
  "targetBundle": "http://full.URL.to.target.bundle",
  "conflicts": {
    "conflict_id1": "http://full.URL.to.operation.outcome",
    "conflict_id2": "http://full.URL.to.operation.outcome"
  }
}
```

Whenever possible I try to refer to bundles and OperationOutcomes using their fully qualified URLs, not just an ID. This saved state format is liable to grow/change as we collect and store more information about a merge session.